lib/workos/types/provider_enum.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+# typed: strict
+
+module WorkOS
+  module Types
+    # The ProviderEnum is type-safe declarations of a
+    # fixed set of values for SSO Providers.
+    class Provider < T::Enum
+      enums do
+        Google = new('Google')
```

Comment:
i took this from the node PR - should it be `GoogleOAuth`?

Reply:
Yeah, I think `GoogleOAuth` makes sense